[SAN-MUYUN] iP

SAN-MUYUN opened the pull request:
# MuyunBot
> "You may delay, but time will not." - Benjamin Franklin [source](https://www.spica.com/blog/the-best-time-management-quotes#:~:text=The%20difference%20between%20successful%20people,Bill%20Gates)

### MuyunBot allows you to keep track of different kinds of tasks ✏️
- **Todo** _just a task with description_
- **Deadline** _task with a deadline_
- **Event** _task with start date and end date_

### MuyunBot also allows you to:
- [x]  **Add tasks** ➕
- [x]  **Delete tasks** ➖ 
- [x]  **Mark tasks as done 🏆  and unmark them later 🔃** 

### Steps to get it running:
1. download MuyunBot.jar
2. use cmd/terminal, navigate to the folder containing MuyunBot.jar
3. run code ``` java -jar MuyunBot.jar```

```java
public class Main {
    public static void main(String[] args) {
        // says hi.
        System.out.println("Happy Coding!!");
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SAN-MUYUN committed:
Change greeting and rename Duke

SAN-MUYUN committed:
Add echo

SAN-MUYUN committed:
Add list and add function

SAN-MUYUN committed:
Adds mark as done and unmark for Level-3

SAN-MUYUN committed:
Add todo, deadline, event classes

SAN-MUYUN committed:
Add UI-testing

SAN-MUYUN committed:
Add error handling

SAN-MUYUN committed:
Add delete function

SAN-MUYUN committed:
Add save function

SAN-MUYUN committed:
Merge branch-Level-7 into master

SAN-MUYUN committed:
Add date time hanlder

SAN-MUYUN committed:
Add more OOP

SAN-MUYUN committed:
Merge branch A-MoreOOP

SAN-MUYUN committed:
Add more OOP

SAN-MUYUN committed:
merging more oop

SAN-MUYUN committed:
Creates packages for different files

SAN-MUYUN committed:
Merge branch 'branch-A-Packages'

SAN-MUYUN committed:
Merge remote-tracking branch 'origin/add-gradle-support'

SAN-MUYUN committed:
Use Gradle

SAN-MUYUN committed:
Add JUnit testing

SAN-MUYUN committed:
Merge branch-A-JUnit

SAN-MUYUN committed:
Build jar

SAN-MUYUN committed:
Add java documentation

SAN-MUYUN committed:
Merge branch with java documentation

SAN-MUYUN committed:
Follow java coding standard

SAN-MUYUN committed:
Add coding standards

SAN-MUYUN committed:
Add find command

SAN-MUYUN committed:
Merge branch with find command implemented

SAN-MUYUN committed:
Resolve merge conflicts

SAN-MUYUN committed:
Add checkstyle and fix coding convention

SAN-MUYUN committed:
Merge branch with checkstyle

Jsjk112344 commented on src/main/java/muyunBot/Storage.java:
```diff
@@ -1,6 +1,6 @@
-package MuyunBot;
+package muyunBot;
```
LGTM! Good understanding of package naming to change to lower case

Jsjk112344 commented on src/main/java/muyunBot/Parser.java:
```diff
@@ -1,12 +1,21 @@
+package muyunBot;
 
-import MuyunBot.exceptions.NoContentException;
+import muyunBot.classes.Deadline;
```
perhaps your muyunBot.classes.Deadline packages should be renamed to a muyunBot.task.Deadline etc

Jsjk112344 commented on src/main/java/muyunBot/Ui.java:
```diff
@@ -1,11 +1,11 @@
 
 public class Ui {
     public static String dashedLines() {
-        return ("    _________"+"\n");
```
perhaps these lines should be declared as final variables

Jsjk112344 commented on src/test/java/muyunBot/ParserTest.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    @Test
+    public void parseDate_invalidMonth_exceptionThrown() {
```
Tests for exceptions, good job!

SAN-MUYUN committed:
Add Javafx GUI

SAN-MUYUN committed:
Add Javafx GUI

SAN-MUYUN committed:
Merge branch 'branch-Level-10'

Mingyuan03 commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```
Comments are nits. Perhaps it might be better to replace `var` with `DialogBox` here for slightly better code readability?

Mingyuan03 commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,71 @@
+    private MuyunBot bot;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
```
In the unlikely scenario that both Images are initialized as `null`, perhaps there can be further exception-handling in the methods for `NullPointerException` too.

Mingyuan03 commented on src/main/java/muyunbot/MuyunBot.java:
```diff
@@ -0,0 +1,73 @@
+
+    /**
+     * Prints greeting message on commandline
+     */
```
Comments are nits here. For methods with a return type, perhaps there can be additional `@return` annotation too!

Mingyuan03 commented on src/test/java/muyunbot/ParserTest.java:
```diff
@@ -0,0 +1,60 @@
+import java.time.format.DateTimeParseException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
```
Static imports could be placed before non-static imports. Genuinely keen to hear second opinions on this. 👍

Mingyuan03 reviewed (commented):
Your code appears on first glance to follow excellent Java coding standards, and that left me thoroughly impressed. Good effort on the detailed documentation!

SAN-MUYUN committed:
Add assert features to document important assumptions

Some codes only run correctly when some invariants are held when as the
code runs.

Add Assertions to classes where important assumptions are made.

Using Assertions ensures that the code is following the correct
invariants while runnning, preventing unexpected behaviors of program.

SAN-MUYUN committed:
Merge pull request #1 from SAN-MUYUN/branch-A-Assertions

Add assert features to document important assumptions

SAN-MUYUN committed:
Improve code quality

Check the code to ensure that the code abides to the code quality given

improve readability of the codes.

SAN-MUYUN committed:
Merge branch 'master' into branch-A-CodeQuality

SAN-MUYUN committed:
Merge pull request #2 from SAN-MUYUN/branch-A-CodeQuality

Improve code quality

SAN-MUYUN committed:
Fix branch conflicts

SAN-MUYUN committed:
Add features to allow update

SAN-MUYUN committed:
Add text bubble for GUI

SAN-MUYUN committed:
Add more testings

SAN-MUYUN committed:
Implement user guide

SAN-MUYUN committed:
Add link to download .jar file

SAN-MUYUN committed:
Fix minor display bug for wrong time format error

SAN-MUYUN committed:
Fix bugs relavant to time in event

SAN-MUYUN committed:
Handle file corruption
